### PR TITLE
Fixing the addressable resolver

### LINKED ISSUE #11: Test for address resolver fails to discover GVK using fake clients - endless loop is occuring
The test:
https://github.com/knative-sandbox/kn-plugin-event/blob/dd15346ae76b9e0c7168d2166ce45c491526a7c4/internal/k8s/addressresolver_test.go#L25-L30

Fails to discover GVK using fake clients. An endless loop is occurring. The unit tests needs to be fixed so that it operated on some fake k8s clients and properly resolve GVK. Running the code vs real clients works well.

### LINKED ISSUE #12: Fails to resolve `duckv1.Addressable` for knative elements
https://github.com/knative-sandbox/kn-plugin-event/blob/dd15346ae76b9e0c7168d2166ce45c491526a7c4/internal/k8s/addressresolver.go#L112-L125

The above code should resolve the found `*unstructured.Unstructured` of knative sinks (like brokers, channels, services) to `duckv1.Addressable` object, but it fails with an error of:

```
Error: reference is invalid: resource isn't addressable: resource isn't addressable: error deserializing duck type *v1.AddressableType into *unstructured.Unstructured error: Object 'Kind' is missing in '{"metadata":{"creationTimestamp":null},"status":{"address":{"url":"http://foo.com"}}}'
```

/kind bug

### PR DESCRIPTION
**Pull request by cardil**
# Changes

- :bug: Fix the addressable resolver for knative components

/kind bug

Fixes #12 
Fixes #11 

**Release Note**

```release-note
From now on kn event plugin now can properly resolve address of targeted knative components.
```

### CODE REVIEW

File: go.mod
Comment (cardil) [inline go.mod:42]: Remove after merge of https://github.com/knative/client/pull/1390

/hold

**Comment by cardil** [comment]
I'll rebase this PR on top of #43

File: go.mod
Comment (cardil) [reply go.mod]: Removed

/unhold

**Review by dsimansk — APPROVED** [review]

**Pull request merged**
